### PR TITLE
Configurator

### PR DESCRIPTION
**Pull request by tinazhouhui**
FINALLY!!

ok this pull request does few things, but MAINLY it implements the configurator component... i had a bit of an issue figuring out the nested object but I GOT IT WORKING! all tests are passing so we are good to goooo!!

also it finally merges the pre commit hook for husky.. yeah yeah should be a separate pull but hey, its only 2 files :)

### CODE REVIEW

File: .husky/custom/block-master.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+branch="$(git rev-parse --abbrev-ref HEAD)"
```
Comment (tinazhouhui) [inline .husky/custom/block-master.sh:1]: This is husky precommit, making sure that if we are on branch master then it does not allow to commit

File: src/components/Configurator.ts [hunk 2]
```diff
@@ -1,42 +1,84 @@
-import path from 'path';
 import fs from 'fs';
 import yaml from 'js-yaml';
 
 export default class Configurator {
 
   path = '';
-  config = {} as any;
+  config: any = {};
```
Comment (tinazhouhui) [inline src/components/Configurator.ts:7]: I dont like the any types, but I really have NO CLUE how to do nested types:
{
  [key: string] : {
     [key: string] : {... } 
  }
}

especially if you dont know how deep you can go

File: tests/components/Configurator.ts [hunk 3]
```diff
@@ -22,12 +22,15 @@ describe('Configurator', () => {
   }
 
   test('should set environment variables', () => {
-    const configurator = new TestConfigurator();
+    const configurator = new TestConfigurator('/../../tests/components/testConfig.yml');
```
Comment (tinazhouhui) [inline tests/components/Configurator.ts:25]: Use a test configurator rather than the actual one :)

File: src/components/Configurator.ts [hunk 4]
```diff
@@ -1,42 +1,85 @@
...
-
-    try {
-      console.log(this.config.parameters.s3);
-    } catch (err: any) {
-      console.error(err.stack);
-    }
+    const envConfig: any = {};
+    for (const [envVariable, configVariable] of Object.entries(this.mapping)) {
+      // check if each key exists in process.env
+      if (process.env[envVariable]) {
```
Comment (Fav8) [inline src/components/Configurator.ts:28]: how could they possibly not exist? You are looping through the mapping at line 9 so it has all of his props. Am I wrong?

**Review by Fav8 — COMMENTED** [review]
we probably need to get in a call because I have no idea what you are trying to achieve with the deepmarge, the code obviously works because I mean it's you 😄

**Review by Fav8 — APPROVED** [review]
good stuff!

**Pull request merged**